src/leader/undirected_graph/yoyo_type_test.go

```diff
@@ -0,0 +1,355 @@
+package undirected_graph
+
+import "testing"
+
+func TestUpdateStatus(t *testing.T) {
+	checkLogOutput()
+
+	s := newStateYoYo(3)
+
+	s.EdgeStates[0] = out
+	s.EdgeStates[1] = out
+	s.EdgeStates[2] = out
+	s.UpdateStatus()
+	if s.Status != source {
+		t.Errorf("Status of vertex with edges (%s,%s,%s) should be source, but is %s",
+			s.EdgeStates[0], s.EdgeStates[1], s.EdgeStates[2], s.Status)
+	}
```

Comment:
I think that whole this code can be extracted to a function with 4 parameters (or even 2, if you pass the whole EdgeState array, not assign one by one)